### DISCUSSION TITLE
Ecommerce - Paymentprovider on order is null

### DISCUSSION THREAD

**Discussion by jonathanelmgren** [pimcore/pimcore#12370] [Q&A / Support]
Hello!

I am trying to create an order with my payment provider, however after the `handleResponse()` function I get thrown an error: `Call to a member function getPaymentProviderKlarna() on null` the error occurs in `OrderAgent.php` - `setPaymentProvider` function.

Where do I set the paymentprovider for the order? The service and config is set up correctly. When I check the order in `OrderAgent.php` it has my order with all the items I ordered but the `paymentProvider` is null.

This is my `startPayment` function
```
    public function startPayment(OrderAgentInterface $orderAgent, PriceInterface $price, AbstractRequest $config): StartPaymentResponseInterface
    {
        //POST to Klarna API and get order object
        $order = $this->createOrder($config);

        $snippet = $order['html_snippet'];

        return new SnippetResponse($orderAgent->getOrder(), $snippet);
    }
```

And this is my `handleResponse()`
```
    public function handleResponse($response)
    {

        //GET order from Klarna API and get order object
        $order = $this->fetchOrder($response);

        return new Status(
            $order['merchant_reference1'],
            $order['order_id'],
            $order['status'],
            $order['status'],
            [
                'klarna_amount' => $order['order_amount'],
                'klarna_htmlsnippet' => $order['html_snippet'],
            ]
        );
    }
```

I am trying to get this working with as minimal config as possible to get it working. But I've been stuck at this point way too long. 

To initiate the payment I run:
`$paymentInformation = $checkoutManager->initOrderPayment();`

And to commit:
`$startPaymentResponse = $checkoutManager->startOrderPaymentWithPaymentProvider($paymentConfig);` (this is where it throws the error)

**Comment by dvesh3** [comment]
@jonathanelmgren are you still facing the issue?

I would first check if the payment provider brick is correctly installed with class link:
![image](https://user-images.githubusercontent.com/5137917/174570430-d2e7761b-2028-4229-aba0-29f0d430ddc9.png)
